### DISCUSSION TITLE
Limit Batch Size

### DISCUSSION THREAD

**Discussion by pedro-santiago** [nova-kit/nova-on-vapor#3] [Q&A]
First of all, excellent job here, I was struggling with this for weeks.

When using, I had trouble when selected resource amount was larger than 15k rows. Managed to add this snippet to mitigate it, might be helpful:

`VaporExportAsCsv.php`
```
        if ($query->count() > 15000) {
            return $response->successful([
                response()->json(static::danger("It's currently not possible to export more than 15.000 rows at once."))
            ]);
        }
```

**Comment by pedro-santiago** [comment]
It's not really an "issue", since it's totally understandable that vapor itself might stumble accross memory limitations. In this scenario, action returns a file with an `Internal Server Error` on it.

**Comment by crynobone** [comment]
Use https://github.com/nova-kit/nova-queued-export-as-csv (loaded in the latest `nova-on-vapor`) if you want to process large resources.
